src/pal/tools/gen-buildsys-clang.sh

```diff
@@ -122,7 +122,7 @@ if [ "$CROSSCOMPILE" == "1" ]; then
     fi
     export TARGET_BUILD_ARCH=$build_arch
     cmake_extra_defines="$cmake_extra_defines -C $CONFIG_DIR/tryrun.cmake"
-    cmake_extra_defines="$cmake_extra_defines -DCMAKE_TOOLCHAIN_FILE=$CONFIG_DIR/toolchain.cmake"
+    cmake_extra_defines="$cmake_extra_defines -DCLR_CMAKE_COMPILER=Clang -DCMAKE_TOOLCHAIN_FILE=$CONFIG_DIR/toolchain.cmake"
```

Comment:
It doesn't work this way, you have basically just duplicated the variable on the command line. You really need to use an environment variable instead.

Reply:
It is strange that we have sysroot but not the other two options we are trying to add:

/usr/bin/clang-5.0 -Wall -std=c11 --sysroot=/crossrootfs/arm -fuse-ld=gold
CMakeFiles/cmTryCompileExec561591157.dir/testCCompiler.c.o -o
cmTryCompileExec561591157 -rdynamic

Where is this -fuse-ld=gold and sysroot arguments coming from?

Reply:
same file? and because env is not set, it is not getting pulled?

Reply:
It is due to the fact that on other than the first usage of the toolchain.cmake, the CLR_CMAKE_COMPILER is not defined, so the CLR_CMAKE_COMPILER STREQUAL "Clang" check doesn't succeed and the options are not added to the command line due to that.

Reply:
got it.